### PR TITLE
chore(docs): Update Vercel hosting information

### PR DESCRIPTION
**Pull request by leerob**
We now support SSR, DSG, and API Routes: https://vercel.com/changelog/improved-support-for-gatsby-sites

### CODE REVIEW

File: docs/docs/how-to/previews-deploys-hosting/deploying-to-vercel.md [hunk 1]
```diff
@@ -2,7 +2,9 @@
 title: Deploying to Vercel
 ---
 
-[Vercel](https://vercel.com/home) is a cloud platform that enables developers to host Jamstack websites and web services that deploy instantly, scale automatically, and requires no supervision, all with zero configuration. They provide a global edge network, SSL encryption, asset compression, cache invalidation, and more.
+[Vercel](https://vercel.com/home) is the platform for frontend developers, providing the speed and reliability innovators need to create at the moment of inspiration. They support over 35 frontend frameworks, including Gatsby.
+
+Vercel supports Gatsby features like SSG, SSR, DSG, API Routes, and more.
```
Comment (LekoArts) [inline docs/docs/how-to/previews-deploys-hosting/deploying-to-vercel.md:7]: Does SSR and DSG support local images/files? (`File` nodes?)

Netlify currently has this limitation (https://github.com/netlify/netlify-plugin-gatsby#caveats) and I checked your source code and I'd assume it's currently not supported. Is that right?

Comment (LekoArts) [reply docs/docs/how-to/previews-deploys-hosting/deploying-to-vercel.md:7]: We mentioned Image CDN below, does Vercel support that? https://www.gatsbyjs.com/blog/image-cdn-lightning-fast-image-processing-for-gatsby-cloud/

Netlify supports it through this: https://github.com/netlify/netlify-plugin-gatsby/blob/main/docs/image-cdn.md

If it doesn't currently it should be mentioned :)

**Review by LekoArts — CHANGES_REQUESTED** [review]
Hi, thanks for the PR! Only two smaller comments.

Thanks for expanding your support on Vercel <3

File: docs/docs/how-to/previews-deploys-hosting/deploying-to-vercel.md [hunk 2]
```diff
@@ -19,8 +21,6 @@ You can use Vercel with Gatsby in multiple ways:
 
 If you want to use a Custom Domain with your Vercel deployment, you can **Add** or **Transfer in** your domain via your Vercel [account Domain settings](https://vercel.com/dashboard/domains).
 
-Vercel doesn't support advanced features like [SSR](/docs/how-to/rendering-options/using-server-side-rendering/), [DSG](/docs/how-to/rendering-options/using-deferred-static-generation/), or [Image CDN](/docs/how-to/images-and-media/using-gatsby-plugin-image/#gatsby-cloud-image-cdn). You can get all features and faster builds by signing up to [Gatsby Cloud](/dashboard/signup).
```
Comment (leerob) [inline docs/docs/how-to/previews-deploys-hosting/deploying-to-vercel.md:22]: Vercel doesn't currently support `File` nodes inside of SSR/DSG, as well as the [Image CDN](/docs/how-to/images-and-media/using-gatsby-plugin-image/#gatsby-cloud-image-cdn).

File: docs/docs/how-to/previews-deploys-hosting/deploying-to-vercel.md [hunk 3]
```diff
@@ -19,8 +21,6 @@ You can use Vercel with Gatsby in multiple ways:
 
 If you want to use a Custom Domain with your Vercel deployment, you can **Add** or **Transfer in** your domain via your Vercel [account Domain settings](https://vercel.com/dashboard/domains).
```
Comment (leerob) [inline docs/docs/how-to/previews-deploys-hosting/deploying-to-vercel.md:23]:
```suggestion

Vercel doesn't currently support `File` nodes inside of SSR/DSG, as well as the [Image CDN](/docs/how-to/images-and-media/using-gatsby-plugin-image/#gatsby-cloud-image-cdn).
```

File: docs/docs/how-to/previews-deploys-hosting/deploying-to-vercel.md [hunk 4]
```diff
@@ -19,8 +21,7 @@ You can use Vercel with Gatsby in multiple ways:
...
 
-Vercel doesn't support advanced features like [SSR](/docs/how-to/rendering-options/using-server-side-rendering/), [DSG](/docs/how-to/rendering-options/using-deferred-static-generation/), or [Image CDN](/docs/how-to/images-and-media/using-gatsby-plugin-image/#gatsby-cloud-image-cdn). You can get all features and faster builds by signing up to [Gatsby Cloud](/dashboard/signup).
-
+Vercel doesn't currently support `File` nodes inside of SSR/DSG, as well as the [Image CDN](/docs/how-to/images-and-media/using-gatsby-plugin-image/#gatsby-cloud-image-cdn).
```
Comment (leerob) [inline docs/docs/how-to/previews-deploys-hosting/deploying-to-vercel.md:24]:
```suggestion
Vercel doesn't currently support `File` nodes inside of SSR/DSG, as well as the [Image CDN](/docs/how-to/images-and-media/using-gatsby-plugin-image/#gatsby-cloud-image-cdn).

```

File: docs/docs/how-to/previews-deploys-hosting/deploying-to-vercel.md (hunk 1 above)
Comment (leerob) [reply docs/docs/how-to/previews-deploys-hosting/deploying-to-vercel.md:7]: Fixed!

**Review by LekoArts — APPROVED** [review]
Thanks!

**Pull request merged**